integration_tests/tests/mod.rs

```diff
@@ -0,0 +1,408 @@
+    /// Which process, if any, had an error? Assumes that exactly one party will error.
+    error: Option<Party>,
+    /// Outcome of channel; left an Option in case we do not expect a channel from a test
+    /// (e.g. interacting with a non-existent channel).
+    channel_outcome: Option<ChannelOutcome>,
+}
+
+#[derive(Debug, Clone)]
+struct ChannelOutcome {
+    customer_status: CustomerStatus,
+    merchant_status: MerchantStatus,
+    customer_balance: CustomerBalance,
+    merchant_balance: MerchantBalance,
+    /// Closing balances of channels, for use in close-related tests
+    closing_balances: Option<ClosingBalances>,
+}
+
+#[derive(Debug, Error)]
+enum TestError {
+    #[error("Operation {0:?} not yet implemented")]
+    NotImplemented(Operation),
+
+    #[error(
+        "The error behavior did not satisfy expected behavior {op:?}. Got
+    CUSTOMER WATCHER OUTPUT:
+    {customer_errors}
+    MERCHANT SERVER OUTPUT:
+    {merchant_errors}
+    OPERATION OUTPUT:
+    {op_error:?}"
+    )]
+    InvalidErrorBehavior {
+        op: Operation,
+        customer_errors: String,
+        merchant_errors: String,
+        op_error: Result<(), anyhow::Error>,
+    },
+
+    #[error("After {op:?}, expected customer status {expected:?}, got {actual:?}")]
+    InvalidCustomerStatus {
+        op: Operation,
+        expected: CustomerStatus,
+        actual: CustomerStatus,
+    },
+
+    #[error("After {op:?}, expected merchant status {expected:?}, got {actual:?}")]
+    InvalidMerchantStatus {
+        op: Operation,
+        expected: MerchantStatus,
+        actual: MerchantStatus,
+    },
+
+    #[error(
+        "After {op:?}, expected customer, merchant balances
+    ({expected_customer:?}, {expected_merchant:?}), got
+    ({actual_customer:?}, {actual_merchant:?})"
+    )]
+    InvalidChannelBalances {
+        op: Operation,
+        expected_customer: CustomerBalance,
+        expected_merchant: MerchantBalance,
+        actual_customer: CustomerBalance,
+        actual_merchant: MerchantBalance,
+    },
+
+    #[error(
+        "After {op:?}, party: {party:?} expected customer, merchant closing balances
+    ({expected_customer:?}, {expected_merchant:?}), got
+    ({actual_customer:?}, {actual_merchant:?})"
+    )]
+    InvalidClosingBalances {
+        op: Operation,
+        party: String,
+        expected_customer: Option<CustomerBalance>,
+        expected_merchant: Option<MerchantBalance>,
+        actual_customer: Option<CustomerBalance>,
+        actual_merchant: Option<MerchantBalance>,
+    },
+}
+
+impl Test {
+    pub async fn execute(
+        &self,
+        customer_config: &customer::Config,
+        merchant_config: &merchant::Config,
+    ) -> Result<(), anyhow::Error> {
+        for (op, expected_outcome) in &self.operations {
+            // Clone inputs. A future refactor should look into having the `Command` trait take
+            // these by reference instead.
+            eprintln!("NEW OP: {:?}", op);
```

Comment:
todo: remove debug statements.

Also, I think the comment above is outdated since we changed the RNG stuff. Please remove that as well.